src/parser.ts

```diff
@@ -365,6 +367,9 @@ export function parseLine(
     case LINE_TYPE.COMPOSITE:
       return parseComposite(input, numberOfComponents, options);
     default:
+      if (admitBareNumbers && lineType === LINE_TYPE.NUMBER) {
```

Comment:
Instead of having `LINE_TYPE.NUMBER` handled with an if statement in the default section of the switch statement here, it might be more consistent to have a case for `LINE_TYPE.NUMBER` just like the other types. The catch is we'd have to handle the case where `admitBareNumbers` is false somehow: that could fall thru to the default case to throw the exception, or (if fall thru is considered unintuitive) there could be a throw in both sections.
 
 TBH, I'd rather have a dispatch table than a switch statement here (although maybe that's just me). What we have here definitely works, and I'm tempted to accept it and just make a note we'd like to refactor this in the future.

Reply:
The parser has been adapted from Scale Workshop 1.5. I think switch statements should be generally avoided, but such refactoring is outside the scope of this PR. I'll make an issue in `scale-workshop-core` as this stuff is migrating there now.